### PR TITLE
Input parser

### PR DESCRIPTION
**Pull request by JaredPotter**
Changes that fix the bugs exposed by Taylor's unit testing in his soon-to-be-merged testing branch.

### CODE REVIEW

File: src/input.c [hunk 1]
```diff
@@ -72,18 +72,24 @@ int input_parser(FILE *fp, UNMOLESTED_INPUT *unin, MOLESTED_INPUT *in)
   if(parse_cnf_header(line, nbvar, nbclauses) != 1)
   { return -1; }
```
Comment (TJHooker33) [inline src/input.c]: oh, hahaha how did we miss that?

File: src/input.c [hunk 2]
```diff
@@ -72,18 +72,24 @@ int input_parser(FILE *fp, UNMOLESTED_INPUT *unin, MOLESTED_INPUT *in)
...
 
-  int** data               = malloc(sizeof(int)*(*nbclauses)*(*nbvar - 1));
+  // Load values into structs.
```
Comment (TJHooker33) [inline src/input.c]: oh, hahaha. how did we miss that?

File: src/input.c [hunk 3]
```diff
@@ -92,8 +98,6 @@ int input_parser(FILE *fp, UNMOLESTED_INPUT *unin, MOLESTED_INPUT *in)
 
   // Load values into structs.
   unin->data = data;
-  unin->nbclauses = *nbclauses;
```
Comment (rolyatwilson) [inline src/input.c:95]: keep all of the unin-> initializations next to each other

File: src/input.c [hunk 4]
```diff
@@ -72,18 +72,24 @@ int input_parser(FILE *fp, UNMOLESTED_INPUT *unin, MOLESTED_INPUT *in)
   if(parse_cnf_header(line, nbvar, nbclauses) != 1)
   { return -1; }
 
-  int** data               = malloc(sizeof(int)*(*nbclauses)*(*nbvar - 1));
+  // Load values into structs.
+  unin->nbclauses = *nbclauses; 
+  unin->nbvars = *nbvar;
+
+  if(*nbvar == 0)
+  { return 1; }
```
Comment (rolyatwilson) [inline src/input.c]: combine line 79, 80 like you did on line 83

File: src/input.c [hunk 5]
```diff
@@ -91,9 +92,13 @@ int input_parser(FILE *fp, UNMOLESTED_INPUT *unin, MOLESTED_INPUT *in)
   { return -1; }
 
   // Load values into structs.
-  unin->data = data;
   unin->nbclauses = *nbclauses; 
   unin->nbvars = *nbvar;
+
+  if(*nbvar == 0) { return 1; }
```
Comment (rolyatwilson) [inline src/input.c]: should this be after line 102

**Pull request merged**